### PR TITLE
Add tsc fixes for report card

### PR DESCRIPTION
**Pull request by prat98**
Typescript fixes to bring reportcard errors to 0

### CODE REVIEW

**Comment by ZitaNemeckova** [comment]
Issues:
- AA 2.1 Onboarding Report no longer shows the chart
- Changes made by job template no longer shows the chart
TSC errors:
```
ERROR in ./src/Containers/Reports/Layouts/Standard/ReportCard.tsx:292:17
TS2322: Type 'string | true | (string | boolean)[]' is not assignable to type 'ChartType | undefined'.
  Type 'string' is not assignable to type 'ChartType | undefined'.
    290 |                 y: chartParams.y,
    291 |                 xTickFormat: chartParams.xTickFormat,
  > 292 |                 chartType: chartParams.chartType,
        |                 ^^^^^^^^^
    293 |               })}
    294 |               dataComponent={'foobar'}
    295 |               data={dataApi.result as any}

ERROR in ./src/Containers/Reports/Layouts/Standard/ReportCard.tsx:344:19
TS2322: Type 'string | true | (string | boolean)[]' is not assignable to type 'ChartType | undefined'.
  Type 'string' is not assignable to type 'ChartType | undefined'.
    342 |                   y: chartParams.y,
    343 |                   xTickFormat: chartParams.xTickFormat,
  > 344 |                   chartType: chartParams.chartType,
        |                   ^^^^^^^^^
    345 |                 })}
    346 |                 dataComponent={'foobar'}
    347 |                 data={dataApi.result as any}

ERROR in ./src/Containers/Reports/Layouts/Standard/ReportCard.tsx:409:15
TS2322: Type 'string | true | (string | boolean)[]' is not assignable to type 'ChartType | undefined'.
  Type 'string' is not assignable to type 'ChartType | undefined'.
    407 |               y: chartParams.y,
    408 |               xTickFormat: chartParams.xTickFormat,
  > 409 |               chartType: chartParams.chartType,
        |               ^^^^^^^^^
    410 |             })}
    411 |             data={dataApi.result as any}
    412 |             specificFunctions={{
```

File: src/Containers/Reports/Layouts/Standard/ReportCard.tsx [hunk 1]
```diff
@@ -56,12 +61,20 @@ const ReportCard: FunctionComponent<StandardProps> = ({
...
-  const { request: fetchData, ...dataApi } = useRequest(readData, {
-    meta: { count: 0, legend: [] },
-  });
+  const { request: fetchData, ...dataApi } = useRequest(
+    readData as unknown as (
+      ...args: unknown[]
+    ) => Promise<NonGroupedApi | GroupedApi>,
+    {
+      meta: { count: 0, legend: [] },
+      items: [],
```
Comment (MilanPospisil) [inline src/Containers/Reports/Layouts/Standard/ReportCard.tsx:75]: This is change to actual code and it is causing problem - in:

https://stage.foo.redhat.com:1337/ansible/automation-analytics/reports?allReports.description=&allReports.limit=20&allReports.name[]&allReports.offset=0&allReports.selected_report=aa_2_1_onboarding&allReports.slug[]&allReports.sort_options=name&allReports.sort_order=asc&allReports.tags[]

No data are rendered while it should render normal data (they are rendering on devel), so this change alter how it behave.

Please do not do any changes in code, rather make typescript stop complaining :-) Its safer to not change any javascript, only typescript.

**Comment by MilanPospisil** [comment]
Great job Pratyush :-) Now lets focus on changes you made in source code, that broke some functionality, we must be very careful with any js changes. TS changes are generaly much more safer.

**Comment by prat98** [comment]
Thanks for the reviews updating right now!

File: src/Containers/Reports/Layouts/Standard/ReportCard.tsx [hunk 2]
```diff
@@ -216,59 +230,69 @@ const ReportCard: FunctionComponent<StandardProps> = ({
...
+      pageLimit={queryParams.limit as any}
+      chartType={chartParams.chartType as string}
       sortOptions={chartParams.y}
-      sortOrder={queryParams.sort_order}
-      dateGranularity={queryParams.granularity}
-      startDate={queryParams.start_date}
-      endDate={queryParams.end_date}
-      dateRange={queryParams.quick_date_range}
-      adoptionRateType={queryParams.adoption_rate_type}
+      sortOrder={
```
[the remark this answers is not in the document]
Comment (prat98) [reply src/Containers/Reports/Layouts/Standard/ReportCard.tsx:245]: I think I made this change before might just be able to typecast to any and it probably should work

**Comment by prat98** [comment]
> Issues:
> 
> * AA 2.1 Onboarding Report no longer shows the chart
> * Changes made by job template no longer shows the chart
> 
> TSC errors:
> 
> ```
> ERROR in ./src/Containers/Reports/Layouts/Standard/ReportCard.tsx:292:17
> TS2322: Type 'string | true | (string | boolean)[]' is not assignable to type 'ChartType | undefined'.
>   Type 'string' is not assignable to type 'ChartType | undefined'.
>     290 |                 y: chartParams.y,
>     291 |                 xTickFormat: chartParams.xTickFormat,
>   > 292 |                 chartType: chartParams.chartType,
>         |                 ^^^^^^^^^
>     293 |               })}
>     294 |               dataComponent={'foobar'}
>     295 |               data={dataApi.result as any}
> 
> ERROR in ./src/Containers/Reports/Layouts/Standard/ReportCard.tsx:344:19
> TS2322: Type 'string | true | (string | boolean)[]' is not assignable to type 'ChartType | undefined'.
>   Type 'string' is not assignable to type 'ChartType | undefined'.
>     342 |                   y: chartParams.y,
>     343 |                   xTickFormat: chartParams.xTickFormat,
>   > 344 |                   chartType: chartParams.chartType,
>         |                   ^^^^^^^^^
>     345 |                 })}
>     346 |                 dataComponent={'foobar'}
>     347 |                 data={dataApi.result as any}
> 
> ERROR in ./src/Containers/Reports/Layouts/Standard/ReportCard.tsx:409:15
> TS2322: Type 'string | true | (string | boolean)[]' is not assignable to type 'ChartType | undefined'.
>   Type 'string' is not assignable to type 'ChartType | undefined'.
>     407 |               y: chartParams.y,
>     408 |               xTickFormat: chartParams.xTickFormat,
>   > 409 |               chartType: chartParams.chartType,
>         |               ^^^^^^^^^
>     410 |             })}
>     411 |             data={dataApi.result as any}
>     412 |             specificFunctions={{
> ```

Odd I didn't see these tsc error on my end. Let me look into them

**Pull request closed without merging**